flink-tests/src/test/java/org/apache/flink/test/recovery/LocalRecoveryITCase.java

```diff
@@ -67,12 +68,14 @@
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /** Tests local recovery by restarting Flink processes. */
 @ExtendWith(TestLoggerExtension.class)
 class LocalRecoveryITCase {
```

Comment:
I think I'm missing something here; how does this test local recovery with the AdaptiveScheduler?

Reply:
When the Adaptive Scheduler is enabled in `enable-adaptive-scheduler` maven [profile](https://github.com/apache/flink/blob/161014149e803bfd1d3653badb230b2ed36ce3cb/pom.xml#L976), it get picked up by [`ClusterOptions`](https://github.com/apache/flink/blob/161014149e803bfd1d3653badb230b2ed36ce3cb/flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java#L226).

This test then kill a TM and verifies that a task get rescheduled onto the TM with the same `AllocationID`.

Reply:
The test wasn't failing because the assertion failed the task, causing a restart; eventually, the right allocation ID would be picked up.
I fixed it in 6c0284ca7bfe09a94740db27db38279234222c63 `[hotfix][tests] Make LocalRecoveryITCase fail when allocations don't match` by checking for any allocationID mismatches (in the main thread).

Reply:
Thanks for the explanation, this makes perfect sense!

I'm trying the test locally, and it could be faster (not because of your changes); it takes 23 seconds on the M1 Mac.

I'm wondering whether we need to start 3 TM subprocesses 🤔. Wouldn't be enough to simply start in-process TMs (MiniCluster)? Wouldn't the AllocationID logic remain unchanged (which is what we test for)?

Reply:
Hmm, it also seems to stress the TM recovery off of the working directory, which currently doesn't seem to be supported by the MiniCluster (the logic is local to the TaskManagerRunner) 🤔

I think it's alright then. Thanks!